h2o-automl/src/main/java/ai/h2o/automl/AutoMLBuildSpec.java

```diff
@@ -174,14 +176,177 @@ public RandomDiscreteValueSearchCriteria getSearchCriteria() {
     public Algo[] exclude_algos;
     public Algo[] include_algos;
     public StepDefinition[] modeling_plan;
+    public final AutoMLCustomParameters algo_parameters = new AutoMLCustomParameters();
   }
 
-  public AutoMLBuildControl build_control;
-  public AutoMLInput input_spec;
-  public AutoMLBuildSpec.AutoMLBuildModels build_models;
+  public static final class AutoMLCustomParameters extends Iced {
+
+    // convenient property to allow us to modify our model (and later grids) definitions
+    // and benchmark them without having to rebuild the backend for each change.
+    private static final String ALGO_PARAMS_ALL_ENABLED = H2O.OptArgs.SYSTEM_PROP_PREFIX + "automl.algo_parameters.all.enabled";
+
+    // let's limit the list of allowed custom parameters by default for now: we can always decide to open this later.
+    private static final String[] ALLOWED_PARAMETERS = {
+            "monotone_constraints",
+            "ntrees",
+    };
+    private static final String ROOT_PARAM = "algo_parameters";
+
+    public static final class AutoMLCustomParameter<V> extends Iced {
+      public AutoMLCustomParameter(String name, V value) {
+        this.name = name;
+        this.value = value;
+      }
+
+      public AutoMLCustomParameter(Algo algo, String name, V value) {
+        this.algo = algo;
+        this.name = name;
+        this.value = value;
+      }
+
+      private Algo algo;
+      private String name;
+      private V value;
+    }
+
+    private final IcedHashMap<String, String[]> algo_parameter_names = new IcedHashMap<>(); // stores the parameters names overridden, by algo name
+    private final IcedHashMap<String, Model.Parameters> algo_parameters = new IcedHashMap<>(); //stores the parameters values, by algo name
+
+    // used for building phase only
+    private final transient List<AutoMLCustomParameter> any_algo_params = new ArrayList<>();
```

Comment:
I know, if there were more scopes, it would make sense.
I considered it was not worth having a Map when the partition is binary: any vs. specific.
Mainly because there's always a bit more logic when you need to manipulate map values.